modules/core/image/src/Plugin/ImageEffect/AutoOrientImageEffect.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\farm_image\Plugin\ImageEffect;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Image\ImageInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\image\Attribute\ImageEffect;
+use Drupal\image\ConfigurableImageEffectBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Automatically rotates an image based on EXIF Orientation.
+ */
+#[ImageEffect(
+  id: "farm_image_auto_orient",
+  label: new TranslatableMarkup("Auto-orient"),
+  description: new TranslatableMarkup("Automatically rotates an image based on EXIF Orientation.")
+)]
+class AutoOrientImageEffect extends ConfigurableImageEffectBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * AutoOrientImageEffect constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   The file system service.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, LoggerInterface $logger, FileSystemInterface $file_system) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $logger);
+    $this->fileSystem = $file_system;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory')->get('image'),
+      $container->get('file_system'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applyEffect(ImageInterface $image) {
+
+    // Only proceed if the exif PHP extension is loaded.
+    if (!extension_loaded('exif')) {
+      return TRUE;
+    }
```

Comment:
Should this check also occur at the top of `transformDimensions`? The [docs](https://api.drupal.org/api/drupal/core%21modules%21image%21src%21ConfigurableImageEffectBase.php/class/ConfigurableImageEffectBase/8.9.x) imply but don't really guarantee that it won't be called if no effect was applied here...